### PR TITLE
add twitter sharing button

### PR DESCRIPTION
**Pull request by boojack**

### CODE REVIEW

**Comment by boojack** [comment]
PTAL

File: index.html [hunk 1]
```diff
@@ -12,6 +12,19 @@
       content="gWDCNMCQeEt3snNQ8J4NCOmVkz4axYeIp5fH4umVaAs"
     />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta name="twitter:card" content="summary_large_image" />
+    <meta name="twitter:site" content="star-history" />
```
Comment (tianzhou) [inline index.html:16]: not sure if this should be star-history.com?

File: index.html [hunk 2]
```diff
@@ -12,6 +12,19 @@
...
+    <meta name="twitter:creator" content="bytebase" />
+    <meta name="twitter:title" content="Star History" />
+    <meta
+      name="twitter:description"
+      content="The missing GitHub star history graph."
+    />
+    <!-- use static image -->
+    <meta
+      name="twitter:image"
+      content="https://star-history.com/star-history.png"
```
Comment (tianzhou) [inline index.html:26]: let's use webp

File: src/components/StarChartViewer.vue [hunk 3]
```diff
@@ -286,22 +303,56 @@ export default defineComponent({
...
+        toast.error("No repo found");
+        return;
+      }
+
+      const starhistoryLink = encodeURIComponent(window.location.href);
+      let text = "";
+
+      if (repos.length === 1) {
+        const repo = repos[0];
+        let starCount = 99;
```
Comment (tianzhou) [inline src/components/StarChartViewer.vue:318]: I feel, if we somehow don't get the count. We just use 0 and don't show the star count text at all

**Review by tianzhou — CHANGES_REQUESTED** [review]

**Review by tianzhou — APPROVED** [review]
LGTM

**Pull request merged**